tremor-codec/src/codec/avro.rs

```diff
@@ -274,20 +306,29 @@ impl Codec for Avro {
         // TODO: application/json-seq for one json doc per line?
     }
 
-    fn decode<'input>(
+    async fn decode<'input>(
         &mut self,
         data: &'input mut [u8],
         _ingest_ns: u64,
-    ) -> Result<Option<Value<'input>>> {
+        meta: Value<'input>,
+    ) -> Result<Option<(Value<'input>, Value<'input>)>> {
         let reader = Reader::new(&*data)?;
 
         let mut vals = reader.map(|v| Self::convert_avro_value(v?));
-        vals.next().transpose()
+        vals.next().map(|v| v.map(|v| (v, meta))).transpose()
     }
 
-    fn encode(&mut self, data: &Value) -> Result<Vec<u8>> {
+    async fn encode(&mut self, data: &Value, meta: &Value) -> Result<Vec<u8>> {
+        let schema = if let Some(schema_id) = meta.get_u32("schema_id") {
```

Comment:
should we also check for `schema_name` here in meta? Otherwise we don't have to store the schemas by name in the `AvroRegistry`, no?

Reply:
schemas are a bit ... odd, when coming from the schema registry they have a id, when they reference each other they have a name 😭